lib/xboxrt/libc_extensions/malloc.h

```diff
@@ -0,0 +1,10 @@
+#ifndef _XBOXRT_MALLOC_H_
+#define _XBOXRT_MALLOC_H_
+
+// Some code expects to find a declaration of malloc here, so we simply include the whole C header
+#include <stdlib.h>
+
+void *_alloca (size_t size);
+static void *alloca (size_t size) { return _alloca(size); }
```

Comment:
Please add a comment to note which standard this extension comes from.

This implementation probably won't work for me, because most code I have seen depends on alloca.h instead of malloc.h; also see Linux man-page:

```
SYNOPSIS
       #include <alloca.h>

       void *alloca(size_t size);

[...]

CONFORMING TO
       This function is not in POSIX.1.

       There is evidence that the alloca() function appeared in 32V, PWB, PWB.2, 3BSD, and 4BSD.  There is a man page
       for it in 4.3BSD.  Linux uses the GNU version.
```

Reply:
It's implemented as described by Microsoft in their C runtime docs. The code is different now, but I added a comment mentioning that `_alloca` is Windows-specific.

Microsoft describes that `_alloca` lives in `malloc.h` (they often don't provide the non-underscored variant by default). We could move the definition of `alloca` to a new `alloca.h` and include that here, if it's a portability issue.